packages/flutter_test/lib/src/event_simulation.dart

```diff
@@ -0,0 +1,350 @@
+    case 'linux':
+      result['toolkit'] = 'glfw';
+      result['keyCode'] = keyCode;
+      result['scanCode'] = scanCode;
+      result['modifiers'] = _getGlfwModifierFlags(key, isDown);
+      break;
+    case 'macos':
+      result['keyCode'] = scanCode;
+      result['characters'] = key.keyLabel;
+      result['charactersIgnoringModifiers'] = key.keyLabel;
+      result['modifiers'] = _getMacOsModifierFlags(key, isDown);
+      break;
+    default:
+      assert(false, 'Platform $platform not supported for key simulation');
+      break;
+  }
+  return result;
+}
+
+int _getAndroidModifierFlags(LogicalKeyboardKey newKey, bool isDown) {
+  int result = 0;
+  final Set<LogicalKeyboardKey> pressed = RawKeyboard.instance.keysPressed;
+  if (isDown) {
+    pressed.add(newKey);
+  } else {
+    pressed.remove(newKey);
+  }
+  if (pressed.contains(LogicalKeyboardKey.shiftLeft)) {
+    result |= RawKeyEventDataAndroid.modifierLeftShift | RawKeyEventDataAndroid.modifierShift;
+  }
+  if (pressed.contains(LogicalKeyboardKey.shiftRight)) {
+    result |= RawKeyEventDataAndroid.modifierRightShift | RawKeyEventDataAndroid.modifierShift;
+  }
+  if (pressed.contains(LogicalKeyboardKey.metaLeft)) {
+    result |= RawKeyEventDataAndroid.modifierLeftMeta | RawKeyEventDataAndroid.modifierMeta;
+  }
+  if (pressed.contains(LogicalKeyboardKey.metaRight)) {
+    result |= RawKeyEventDataAndroid.modifierRightMeta | RawKeyEventDataAndroid.modifierMeta;
+  }
+  if (pressed.contains(LogicalKeyboardKey.controlLeft)) {
+    result |= RawKeyEventDataAndroid.modifierLeftControl | RawKeyEventDataAndroid.modifierControl;
+  }
+  if (pressed.contains(LogicalKeyboardKey.controlRight)) {
+    result |= RawKeyEventDataAndroid.modifierRightControl | RawKeyEventDataAndroid.modifierControl;
+  }
+  if (pressed.contains(LogicalKeyboardKey.altLeft)) {
+    result |= RawKeyEventDataAndroid.modifierLeftAlt | RawKeyEventDataAndroid.modifierAlt;
+  }
+  if (pressed.contains(LogicalKeyboardKey.altRight)) {
+    result |= RawKeyEventDataAndroid.modifierRightAlt | RawKeyEventDataAndroid.modifierAlt;
+  }
+  if (pressed.contains(LogicalKeyboardKey.fn)) {
+    result |= RawKeyEventDataAndroid.modifierFunction;
+  }
+  if (pressed.contains(LogicalKeyboardKey.scrollLock)) {
+    result |= RawKeyEventDataAndroid.modifierScrollLock;
+  }
+  if (pressed.contains(LogicalKeyboardKey.numLock)) {
+    result |= RawKeyEventDataAndroid.modifierNumLock;
+  }
+  if (pressed.contains(LogicalKeyboardKey.capsLock)) {
+    result |= RawKeyEventDataAndroid.modifierCapsLock;
+  }
+  return result;
+}
+
+int _getGlfwModifierFlags(LogicalKeyboardKey newKey, bool isDown) {
+  int result = 0;
+  final Set<LogicalKeyboardKey> pressed = RawKeyboard.instance.keysPressed;
+  if (isDown) {
+    pressed.add(newKey);
+  } else {
+    pressed.remove(newKey);
+  }
+  if (pressed.contains(LogicalKeyboardKey.shiftLeft) || pressed.contains(LogicalKeyboardKey.shift) || pressed.contains(LogicalKeyboardKey.shiftRight)) {
```

Comment:
is it possible to have LogicalKeyboardKey.shift given it is transformed to left version of it in _getSynonym?

Reply:
Yeah. probably not. I coded that before I added the synonym code.  Fixed.